src/main/java/com/google/research/bleth/simulator/Agent.java

```diff
@@ -0,0 +1,11 @@
+package com.google.research.bleth.simulator;
+
+/** Represent both observer and beacon. */
+public interface Agent {
```

Comment:
Consider adding a getLocation() method so the agent can report its existing location.